src/fft.js

```diff
@@ -467,9 +467,7 @@ define(function (require) {
    *                               Defaults to 0.8.
    */
   p5.FFT.prototype.smooth = function(s) {
-    if (s) {
-      this.smoothing = s;
-    }
+    this.smoothing = s;
     this.analyser.smoothingTimeConstant = s;
```

Comment:
It is possible for `this.smoothing`  to be set even if `analyser.smoothingTimeConstant` fails.

One option is to switch the order.

But a better approach might be to to use a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) for smoothing in the constructor of p5.FFT. What do you think of like this?

```
Object.defineProperty(this, 'smoothing', {
  get: function() {
    return this.fft.smoothingTimeConstant
  }
});
```